internal/bundler/bundler_test.go

```diff
@@ -1386,6 +1386,72 @@ console.log(main_esm_default());
 	})
 }
 
+func TestTsConfigPaths(t *testing.T) {
+	expectBundled(t, bundled{
+		files: map[string]string{
+			"/Users/user/project/src/entry.js": `
+				import fn from 'core/test'
+				import fn2 from 'testing/test'
```

Comment:
It appears this case isn't actually supposed to work. At least the TypeScript compiler throws an error here. I can do the testing work and make adjustments to the algorithm as appropriate.